app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadService.java

```diff
@@ -496,16 +496,16 @@ private void updateForeground() {
         if (downloadStatus.get(chunk.getNotificationID()) == CANCEL) {
           String path = file.getPath();
           Log.i(KIWIX_TAG, "Download Cancelled, deleting file: " + path);
-          if (path.substring(path.length() - 8).equals("zim.part")) {
-            path = path.substring(0, path.length() - 5);
+          if (path.substring(path.length() - 13).equals("zim.part.part")) {
+            path = path.substring(0, path.length() - 10);
             FileUtils.deleteZimFile(path);
           } else {
-            path = path.substring(0, path.length() - 7) + "aa";
+            path = path.substring(0, path.length() - 12) + "aa";
             FileUtils.deleteZimFile(path);
           }
         } else {
           Log.i(KIWIX_TAG, "Download completed, renaming file ([" + file.getPath() + "] -> .zim)");
-          file.renameTo(new File(file.getPath().replace(".part", "")));
+          file.renameTo(new File(file.getPath().replace(".part.part", "")));
```

Comment:
I don't think you have understood the task. From what I can see you have simply renamed .part to .part.part with no semantic difference. The code at this line should only remove one .part and once the total file has been downloaded all chunks should be renamed to remove the last .part.